packages/@aws-cdk/aws-iam/lib/oidc-provider/external.ts

```diff
@@ -20,26 +21,77 @@ function defaultLogger(fmt: string, ...args: any[]) {
 /**
  * Downloads the CA thumbprint from the issuer URL
  */
-async function downloadThumbprint(issuerUrl: string) {
-  external.log(`downloading certificate authority thumbprint for ${issuerUrl}`);
+export async function downloadThumbprint(issuerUrl: string) {
+
+  external.log(`Downloading certificate authority thumbprint for ${issuerUrl}`);
+
   return new Promise<string>((ok, ko) => {
     const purl = url.parse(issuerUrl);
     const port = purl.port ? parseInt(purl.port, 10) : 443;
+
     if (!purl.host) {
       return ko(new Error(`unable to determine host from issuer url ${issuerUrl}`));
     }
+
     const socket = tls.connect(port, purl.host, { rejectUnauthorized: false, servername: purl.host });
     socket.once('error', ko);
+
     socket.once('secureConnect', () => {
-      const cert = socket.getPeerCertificate();
+      // This set to `true` would return the entire chain of certificates as a circular reference object
+      let cert = socket.getPeerCertificate(true);
+
+      const unqiueCerts = new Set<DetailedPeerCertificate>();
+      do {
+        unqiueCerts.add(cert);
+        cert = cert.issuerCertificate;
+      } while ( cert && typeof cert === 'object' && !unqiueCerts.has(cert));
+
+      // The last `cert` obtained must be the root certificate in the certificate chain
+      const rootCert = [...unqiueCerts].pop()!;
```

Comment:
is there any case in which `uniqueCerts` is empty?

Reply:
I believe it should not happen in our scenario since this is OIDC. But if the user passes in non-secure website url then that probably would not have a certificate.

I will add a check for it. But if the website does not have a certificate, we should error out since that is not a secure website.